lib/charms/sdcore_gnbsim/v0/fiveg_gnb_identity.py

```diff
@@ -0,0 +1,290 @@
+
+        def _on_fiveg_gnb_identity_request(self, event):
+            ...
+            self.gnb_identity_provider.publish_gnb_identity_information(
+                relation_id=event.relation_id,
+                gnb_name=name,
+                tac=tac,
+            )
+    ```
+
+    And a corresponding section in charm's `metadata.yaml`:
+    ```
+    provides:
+        fiveg_gnb_identity:  # Relation name
+            interface: fiveg_gnb_identity  # Relation interface
+    ```
+
+Charms that require the `fiveg_gnb_identity` relation should use `GnbIdentityRequires`.
+Typical usage of this class would look something like:
+
+    ```python
+    ...
+    from charms.sdcore_gnbsim.v0.fiveg_gnb_identity import GnbIdentityRequires
+    ...
+
+    class SomeRequirerCharm(CharmBase):
+
+        def __init__(self, *args):
+            ...
+            self.fiveg_gnb_identity = GnbIdentityRequires(charm=self, relation_name="fiveg_gnb_identity")
+            ...
+            self.framework.observe(self.fiveg_gnb_identity.on.fiveg_gnb_identity_available, 
+                self._on_fiveg_gnb_identity_available)
+
+        def _on_fiveg_gnb_identity_available(self, event):
+            gnb_name = event.gnb_name,
+            tac = event.tac,
+            # Do something with the gnB's name and TAC.
+    ```
+
+    And a corresponding section in charm's `metadata.yaml`:
+    ```
+    requires:
+        fiveg_gnb_identity:  # Relation name
+            interface: fiveg_gnb_identity  # Relation interface
+    ```
+"""
+
+import logging
+
+from interface_tester.schema_base import DataBagSchema  # type: ignore[import]
+from ops.charm import CharmBase, CharmEvents, RelationChangedEvent, RelationJoinedEvent
+from ops.framework import EventBase, EventSource, Object
+from pydantic import BaseModel, Field, ValidationError
+
+# The unique Charmhub library identifier, never change it
+LIBID = "ca9a66c5806e47e7b2750e8cdf696b80"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+PYDEPS = ["pydantic", "pytest-interface-tester"]
+
+
+logger = logging.getLogger(__name__)
+
+"""Schemas definition for the provider and requirer sides of the `fiveg_gnb_identity` interface.
+It exposes two interfaces.schema_base.DataBagSchema subclasses called:
+- ProviderSchema
+- RequirerSchema
+
+Examples:
+    ProviderSchema:
+        unit: <empty>
+        app: {
+            "gnb_name": "gnb001",
+            "tac": "00001"
+        }
+    RequirerSchema:
+        unit: <empty>
+        app:  <empty>
+"""
+
+
+class FivegGnbIdentityProviderAppData(BaseModel):
+    """Provider app data for fiveg_gnb_identity."""
+
+    gnb_name: str = Field(
+        description="Name of the gnB.",
+        examples=["gnb001"]
+    )
+    tac: str = Field(
```

Comment:
Any reason why we want TAC to be a string instead of an integer?

Reply:
in the `sdcore-gnbsim-operator` config the TAC is a string and its default value is 000001

Reply:
I also thought it was an integer when I read the spec but then I found this on the implementation.
Also, I think that the TAC is supposed to be hex number ?

Reply:
I'm actually not sure about it. In extended version it can be represented by a 3-octet string, but essentially that's just a representation of a number from 0 to 16 million something.
The reason I'm asking this is because when we [configure network slices](https://canonical-charmed-5g.readthedocs-hosted.com/en/latest/tutorial/03_configure_sdcore/), we use `int`. Did you already try testing the whole thing together?

Reply:
oh I see. Indeed we can have some inconsistencies here. 
I haven't tried the whole thing together yet

Reply:
OK, I verified this and here's the result:
```shell
ubuntu@ip-172-31-32-116:~$ curl -v ${WEBUI_IP}:5000/config/v1/network-slice/default \
--header 'Content-Type: application/json' \
--data '{
  "slice-id": {
    "sst": "1",
    "sd": "010203"
  },
  "site-device-group": [
    "cows"
  ],
  "site-info": {
    "site-name": "demo",
    "plmn": {
      "mcc": "208",
      "mnc": "93"
    },
    "gNodeBs": [
      {
        "name": "demo-gnb1",
        "tac": "00001"
      }
    ],
    "upf": {
      "upf-name": "upf-external",
      "upf-port": "8805"
    }
  }
}'
*   Trying 10.1.80.45:5000...
* Connected to 10.1.80.45 (10.1.80.45) port 5000 (#0)
(...)
< HTTP/1.1 400 Bad Request
(...)
```
And indeed, the slice doesn't get created:
```shell
ubuntu@ip-172-31-32-116:~$ curl -X GET http://${WEBUI_IP}:5000/config/v1/network-slice/default
null
ubuntu@ip-172-31-32-116:~$ curl -X GET http://${WEBUI_IP}:5000/config/v1/network-slice
[]
```
When I change TAC to `int`, it works as expected.